### PR TITLE
Feature/project code & short names

### PR DESCRIPTION
**Pull request by scottmcdonnell**
**Problem**
We use project codes and short names in our workfile naming conventions and wanted to view and edit these in Kitsu to manage short names.

These short_names are also synced with Ayon and can be used there in the Pipeline.

Example:
- Project: Test Project -> TP
- Asset Type: Prop -> PROP
- Task: Modelling -> MDL

Our workfiles can be shortened to TP_PROP_MDL_NAME_V01.ma

**Solution**
Added project.code to UI - renamed as Short Name in local for consistency.

Project List:
image

Project Edit:
image

Added short_name to the lists and edits + CSV export.

Task Types List:
image

Asset Type List:
image

Asset Type short_name relies on PR - https://github.com/cgwire/zou/pull/823

### CODE REVIEW

**Comment by EvanBldy** [comment]
Hi @scottmcdonnell, 
Thanks for your PRs, we will merge them this week (we need to do a release before).

File: src/components/pages/production/ProductionTaskTypes.vue [hunk 1]
```diff
@@ -48,6 +48,16 @@
...
+              /*
+              <thead>
+                <tr>
+                  <th class="th-name">{{ $t('task_status.fields.name') }}</th>
+                  <th class="th-short-name">
+                    {{ $t('task_status.fields.short_name') }}
+                  </th>
+                </tr>
+              </thead>
+              */
```
Comment (NicoPennec) [inline src/components/pages/production/ProductionTaskTypes.vue:60]: invalid comment in the template.
```suggestion
              <!--
              <thead>
                <tr>
                  <th class="th-name">{{ $t('task_status.fields.name') }}</th>
                  <th class="th-short-name">
                    {{ $t('task_status.fields.short_name') }}
                  </th>
                </tr>
              </thead>
              -->
```

File: src/components/modals/EditAssetTypeModal.vue [hunk 2]
```diff
@@ -25,7 +25,14 @@
...
             v-focus
           />
-
+          <text-field
+            ref="shortNameField"
+            :label="$t('asset_types.fields.short_name')"
+            :maxlength="30"
+            v-model="form.short_name"
+            @enter="runConfirmation"
+            v-focus
```
Comment (NicoPennec) [inline src/components/modals/EditAssetTypeModal.vue:34]: useless directive.
`v-focus` is already set on the name field.

```suggestion
```

File: src/components/modals/EditProductionModal.vue [hunk 3]
```diff
@@ -24,6 +24,13 @@
             @enter="runConfirmation"
             v-focus
           />
+          <text-field
+            ref="codeField"
+            :label="$t('productions.fields.code')"
+            v-model="form.code"
+            @enter="runConfirmation"
+            v-focus
```
Comment (NicoPennec) [inline src/components/modals/EditProductionModal.vue:32]: useless directive.
`v-focus` is already set on the name field.

```suggestion
```

File: src/components/modals/EditTaskTypeModal.vue [hunk 4]
```diff
@@ -24,6 +24,13 @@
             @enter="confirmClicked"
             v-focus
           />
+          <text-field
+            ref="shortNameField"
+            :label="$t('task_types.fields.short_name')"
+            v-model="form.short_name"
+            @enter="confirmClicked"
+            v-focus
```
Comment (NicoPennec) [inline src/components/modals/EditTaskTypeModal.vue:32]: useless directive.
`v-focus` is already set on the name field.

```suggestion
```

File: src/components/pages/production/ProductionTaskType.vue [hunk 5]
```diff
@@ -1,6 +1,9 @@
 <template>
   <tr class="datatable-row" :key="taskType.id">
     <task-type-cell :task-type="taskType" />
+    <td class="name">
```
Comment (NicoPennec) [inline src/components/pages/production/ProductionTaskType.vue:4]: rename class

```suggestion
    <td class="short-name">
```

**Review by NicoPennec — APPROVED** [review]

**Comment by NicoPennec** [comment]
@scottmcdonnell
Thanks for your contribution 🙏 
It will be available in the next release!

**Pull request merged**